gprofiler/metadata/application_identifiers.py

```diff
@@ -133,49 +134,53 @@ def get_application_name(self, process: Process) -> Optional[str]:
 
 
 class _UwsgiApplicationIdentifier(_ApplicationIdentifier):
+    # separated so that we can mock it easily in the tests
     @staticmethod
-    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
-        for arg in process.cmdline():
-            if arg.endswith(".ini"):
-                config_file = arg
-                break
-        else:
-            return None
+    def _open_uwsgi_config_file(process: Process, config_file: str) -> TextIO:
+        return open(resolve_host_path(process, os.path.join(process.cwd(), config_file)))
+
+    @classmethod
+    def _find_wsgi_from_config_file(cls, process: Process) -> Tuple[Optional[str], Optional[str]]:
+        cmdline = process.cmdline()
 
-        if not os.path.isabs(config_file):
-            config_file = os.path.join(process.cwd(), config_file)
+        config_file = _get_cli_arg_by_name(cmdline, "--ini", check_for_equals_arg=True)
+        if config_file is _NON_AVAILABLE_ARG:
+            config_file = _get_cli_arg_by_name(cmdline, "--ini-paste", check_for_equals_arg=True)
+
+        if config_file is _NON_AVAILABLE_ARG:
+            return None, None
 
         config = configparser.ConfigParser(strict=False)
-        config.read(resolve_host_path(process, config_file))
+        config.read_file(cls._open_uwsgi_config_file(process, config_file))
```

Comment:
Close the file in `finally` block (or either use some contextmanager on your `_open_uwsgi_config_file`) 
I know you're gonna tell me GC will clean it but it's a bad practice.